File: backend/src/main/java/com/example/attendance/entity/Employee.java
package com.example.attendance.entity;

import com.example.attendance.enums.Role;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "employees")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 255)
    private String email;

    @Column(nullable = false, length = 255)
    private String password;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Role role;

    @Column(length = 100)
    private String position;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;

    @Column(name = "paid_leave_balance", nullable = false)
    @Builder.Default
    private int paidLeaveBalance = 20;

    @Version
    @Column(nullable = false)
    private Long version;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/employee.ts
export interface EmployeeResponse {
  id: number;
  name: string;
  email: string;
  departmentId: number;
  departmentName: string;
  role: string;
  position: string | null;
  active: boolean;
  paidLeaveBalance: number;
}

export interface LoginRequest {
  employeeId: number;
  password: string;
}

export interface LoginResponse {
  employee: EmployeeResponse;
}

export interface EmployeeCreateRequest {
  name: string;
  email: string;
  departmentId: number;
  role: string;
  position?: string;
  password: string;
}

export interface EmployeeUpdateRequest {
  name: string;
  email: string;
  departmentId: number;
  role: string;
  position?: string;
}

export interface PasswordResetRequest {
  newPassword: string;
}
